db/compaction/compaction_picker_level.cc

```diff
@@ -281,7 +281,13 @@ void LevelCompactionBuilder::SetupInitialFiles() {
   }
 
   // Periodic Compaction
-  PickFileToCompact(vstorage_->FilesMarkedForPeriodicCompaction(), false);
+  bool periodic_compact_to_next_level =
+      ioptions_.level_compaction_dynamic_level_bytes &&
+      (mutable_cf_options_.ttl == 0 /* ttl was disabled */ ||
+       mutable_cf_options_.ttl >
+           mutable_cf_options_.periodic_compaction_seconds);
+  PickFileToCompact(vstorage_->FilesMarkedForPeriodicCompaction(),
```

Comment:
If a file in the last level is marked for periodic compaction, we expect it to be compact and to the same level. May need to update `PickFileToCompact` for that.